lib/attack.go

```diff
@@ -204,6 +204,16 @@ func MaxBody(n int64) func(*Attacker) {
 	return func(a *Attacker) { a.maxBody = n }
 }
 
+// Client returns a functional option that allows you to BYO http.Client
+func Client(c http.Client) func(*Attacker) {
+	return func(a *Attacker) { a.client = c }
+}
+
+// Dialer returns a functional option that allows you to BYO *net.Dialer
+func Dialer(d *net.Dialer) func(*Attacker) {
```

Comment:
Given we already allow setting the `http.Client` above, do we need this option? You can set the dialer in the client directly.